CONTRIBUTING.md

```diff
@@ -0,0 +1,166 @@
+
+> [!NOTE]
+> Once a maintainer reviews your code, please address feedback without rebasing when possible.
+> This includes [synchronizing your PR](https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch)
+> with `main`. The GitHub review experience is much nicer with traditional merge commits.
+
+## Changelog and versioning
+
+**This work is done by the maintainers exclusively.**
+
+In order to generate a new release, it is recommended to use the commands:
+
+
+```
+npm run release:minor
+npm run release:patch
+npm run release:major
+```
+
+
+This includes all the changes in the [CHANGELOG](./CHANGELOG.md) and ensures that the `package.json` and `package-lock.json` are up to date.
+
+You can discard the tag that has been generated locally, as we won't use it.
+
+## Releasing a new version
+
+**This work is done by the maintainers exclusively.**
+
+It is important to ensure that the `package.json`, `package-lock.json` and `CHANGELOG.md` are correct and include all the details for the new release in the `main` branch.
+
+In order to create a new release, follow these steps:
+
+1. Use the GitHub web UI for [new releases](https://github.com/ossf/scorecard-monitor/releases/new).
+2. Mark `Publish this Action to the GitHub Marketplace` as we want to deliver this to our users.
+3. Target the new release version, like `v.1.0.3-beta5`. Note that you can use metadata like `-beta5` and you must include `v` as prefix.
+4. Mark `Set as the latest release`
+5. (Optionally) mark `Set as a pre-release` if it is non-production ready.
+6. :bulb: **Hint:** You can check another release ([example](https://github.com/ossf/scorecard-monitor/releases/tag/v2.0.0-beta7)) to follow the style for title and description (_Main Changes, PRs, New contributions_). 
+7. Click the `Generate release notes` button in the top right to automatically populate the release description.
+
+## What to do before submitting a pull request
+
+The following are the targets that can be used to test your changes locally:
+
+| Command  | Description                                        | Is called in the CI? |
+| -------- | -------------------------------------------------- | -------------------- |
+| `npm run lint:fix` | force the JS Standard style on your files | checked as `npm run lint` |
+| `npm run test:coverage` | Run the unit tests | Yes |
+| `npm run build` | Use ncc to generate the `dist/` folder | yes, it has a validation step |
+
+## Where the CI Tests are configured
+
+1.  See the [action files](.github/workflows) to check its tests, and the
+    scripts used on it.
+
+## Updating Docs
+
+The documentation can be found in the [README](./README.md). Any changes that are merged to `main` will be reflected directly on https://github.com/marketplace/actions/openssf-scorecard-monitor. Therefore, documentation changes do not require a specific release.
```

Comment:
All of these sections should precede or exist as part of the section on opening pull requests.

Reply:
Great suggestion! Done in https://github.com/ossf/scorecard-monitor/pull/85/commits/d2068e47d1738d1f23bb65077ea918bcbb803704